src/System.Net.Sockets/src/System/Net/Sockets/OverlappedAsyncResult.Windows.cs

```diff
@@ -42,11 +41,11 @@ internal unsafe int GetSocketAddressSize()
         // These calls are outside the runtime and are unmanaged code, so we need
         // to prepare specific structures and ints that lie in unmanaged memory
         // since the overlapped calls may complete asynchronously.
-        internal void SetUnmanagedStructures(byte[] buffer, int offset, int size, Internals.SocketAddress socketAddress, bool pinSocketAddress)
+        internal void SetUnmanagedStructures(byte[] buffer, Internals.SocketAddress socketAddress)
         {
             // Fill in Buffer Array structure that will be used for our send/recv Buffer
             _socketAddress = socketAddress;
-            if (pinSocketAddress && _socketAddress != null)
+            if (_socketAddress != null)
             {
                 object[] objectsToPin = null;
                 objectsToPin = new object[2];
```

Comment:
Do we need to reallocate this array every time SetUnmanagedStructures is called, or could we use the previous one if one exists and it's big enough?